data/src/main/java/com/architect/coders/mu8/data/remote/repository/ComicRepository.kt

```diff
@@ -1,18 +1,20 @@
 package com.architect.coders.mu8.data.remote.repository
 
-import com.architect.coders.mu8.data.MARVEL_PUBLIC_KEY
-import com.architect.coders.mu8.data.createHashCode
 import com.architect.coders.mu8.data.remote.RetrofitManager
+import com.architect.coders.mu8.data.remote.mapper.ComicMapper
+import com.architect.codes.mu8.callback.IComicRepository
+import com.architect.codes.mu8.model.Comic
 
-class ComicRepository :
-    IComicRepository {
+class ComicRepository(private val mapper: ComicMapper = ComicMapper()) : IComicRepository {
     private val retrofitManager = RetrofitManager()
+    private val service = retrofitManager.getService()
 
-    override suspend fun getAllComics(timeStamp: String) =
-        retrofitManager.getService()
-            .getAllComics(
-                timeStamp = timeStamp,
-                apiKey = MARVEL_PUBLIC_KEY,
-                hash = createHashCode()
-            )
-}
+    override suspend fun getAllComics(): List<Comic> {
+        val response = service.getAllComics()
+
+        response.body()?.apply {
+            return data.results.map { comicR -> mapper.transform(comicR) }
```

Comment:
What is `comicR`? Why not just use `comic` or the complete `comicsResponse`?

Reply:
done comicsResponse